packages/remark-grid-tables/src/index.js

```diff
@@ -0,0 +1,469 @@
+      if (!isSeparationLine(partLine)) {
+        cell._lines.push(line.substring(cell._startPosition, cell._endPosition))
+        cell._rowspan += 1
+        remainingCells.push(cell)
+      }
+    }
+    // Generate new row
+    this.addRow()
+    const newCells = []
+    for (let c = 0; c < remainingCells.length; ++c) {
+      const remainingCell = remainingCells[c]
+      for (let cc = 0; cc < this.lastRow()._cells.length; ++cc) {
+        const cell = this.lastRow()._cells[cc]
+        if (cell._endPosition < remainingCell._startPosition && newCells.indexOf(cell) === -1) {
+          newCells.push(cell)
+        }
+      }
+      newCells.push(remainingCell)
+      for (let cc = 0; cc < this.lastRow()._cells.length; ++cc) {
+        const cell = this.lastRow()._cells[cc]
+        if (cell._startPosition > remainingCell._endPosition && newCells.indexOf(cell) === -1) {
+          newCells.push(cell)
+        }
+      }
+    }
+
+    // Remove duplicates
+    for (let nc = 0; nc < newCells.length; ++nc) {
+      let newCell = newCells[nc]
+      for (let ncc = 0; ncc < newCells.length; ++ncc) {
+        if (nc !== ncc) {
+          const other = newCells[ncc]
+          if (other._startPosition >= newCell._startPosition &&
+          other._endPosition <= newCell._endPosition) {
+            if (other._lines.length === 0) {
+              newCells.splice(ncc, 1)
+              ncc -= 1
+              if (nc > ncc) {
+                nc -= 1
+                newCell = newCells[nc]
+              }
+            }
+          }
+        }
+      }
+    }
+    this.lastRow()._cells = newCells
+  }
+
+  let TableRow = function (linesInfos) {
+    this._linesInfos = linesInfos
+    this._cells = []
+    for (let i = 0; i < linesInfos.length - 1; ++i) {
+      this._cells.push(new TableCell(linesInfos[i] + 1, linesInfos[i + 1]))
+    }
+  }
+
+  TableRow.prototype.updateContent = function (line) {
+    for (let c = 0; c < this._cells.length; ++c) {
+      const cell = this._cells[c]
+      cell._lines.push(line.substring(cell._startPosition, cell._endPosition))
+    }
+  }
+
+  let TableCell = function (startPosition, endPosition) {
+    this._startPosition = startPosition
+    this._endPosition = endPosition
+    this._colspan = 1
+    this._rowspan = 1
+    this._lines = []
+  }
+
+  TableCell.prototype.mergeWith = function (other) {
+    this._endPosition = other._endPosition
+    this._colspan += other._colspan
+    const newLines = []
+    for (let l = 0; l < this._lines.length; ++l) {
+      newLines.push(`${this._lines[l]}|${other._lines[l]}`)
+    }
+    this._lines = newLines
+  }
+
+  function extractTableContent (lines, linesInfos, hasHeader) {
+    const table = new Table(linesInfos)
+
+    for (let l = 0; l < lines.length; ++l) {
+      const line = lines[l]
+      const matchHeader = hasHeader & isHeaderLine(line) !== null
+      const isEndLine = matchHeader | isPartLine(line) !== null
```

Comment:
Please add a comment explaining the use of these two bitwise operators (or refactor them to make it clear that you're actually working with booleans)